### PR TITLE
Organizational Refactoring, Improved Typing, and Observable Integration in Blog Services

### PR DESCRIPTION
**Pull request by jhonacs2**
# Angular App version

- [ ] `angular-v17` - has no UI libraries
- [ ] `angular-v17-modMAT` - modules and Angular Material
- [ ] `angular-v17-AnguMAT` - has Angular Material
- [x] `angular-v17-PrimeNG` - has PrimeNG

## Description

Refactored folder structure for better organization, implemented typing in blog services for improved code clarity, and transitioned posts.component.ts to utilize observables instead of signals.

## What type of PR is this? (check all applicable)
- [ ] 🅰️ Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Added to documentation?
- [ ] 📜 README.md
- [x ] 🙅 no documentation needed

### CODE REVIEW

**Comment by monacodelisa** [comment]
@czBalazs98 I would like to hear your thoughts since you worked on those files before, and I am conflicted - I have used observables more so I lean towards them, but would be nice to hear your point of view

**Comment by cleancodecraft** [comment]
As far as I can see - a signal is used just for displaying the post-details page, matched by slug, for that I think maybe signal is fine ?

**Comment by monacodelisa** [comment]
> As far as I can see - a signal is used just for displaying the post-details page, matched by slug, for that I think maybe signal is fine ?

I want to use the details page for both - posts from the `posts` component and from `series` component, I am trying to think about this but it has been a very long day, I will have to revisit it tomorrow

File: angular-v17-PrimeNG/src/app/components/post-detail/post-detail.component.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    TagModule,
+    AsyncPipe
+  ],
+  templateUrl: './post-detail.component.html',
+  styleUrl: './post-detail.component.scss'
+})
+export class PostDetailComponent {
+  @Input() slug!: string;
+
+  post = new Observable<Post>();
```
Comment (monacodelisa) [inline angular-v17-PrimeNG/src/app/components/post-detail/post-detail.component.ts:22]: @jhonacs2 let's prepare it for merging, cloud you change with `post$!: Observable<Post>;` and resolve the conflicts and then I can merge, thank you

**Review by cleancodecraft — APPROVED** [review]

**Review by yshashi — APPROVED** [review]

**Comment by czBalazs98** [comment]
> @czBalazs98 I would like to hear your thoughts since you worked on those files before, and I am conflicted - I have used observables more so I lean towards them, but for something like this signals work great and are faster, would be nice to hear your point of view, how do you find the changes ?

Sorry for the late reply, I see that you choose observables, that's perfectly fine for me. I like using signals because it's a kinda new thing and it's nice to try it out, also in my opinion it's a better practice not to depend on third-party libraries like RxJS. But it's just a matter of taste, observables are perfectly fine for this purpose. :)

**Comment by monacodelisa** [comment]
> > @czBalazs98 I would like to hear your thoughts since you worked on those files before, and I am conflicted - I have used observables more so I lean towards them, but for something like this signals work great and are faster, would be nice to hear your point of view, how do you find the changes ?
> 
> Sorry for the late reply, I see that you choose observables, that's perfectly fine for me. I like using signals because it's kinda a new thing and it's nice to try it out, also in my opinion it's a better practice not to depend on third-party libraries like RxJS when there's a solution in Angular itself for the same purpose. But it's just a matter of taste, observables are perfectly fine for this purpose. :)

@czBalazs98 observables for now because it makes it easier to move further along and display post details for posts in series too, but it can be refractured and optimized further along

**Pull request merged**